src/sqliteformat.cpp

```diff
@@ -366,6 +474,55 @@ bool SqliteFormat::purgeDeletedComponents(const KCalendarCore::Incidence &incide
     return false;
 }
 
+bool SqliteFormat::purgeAllComponents(const QString &notebook)
+{
+    if (notebook.isEmpty()) {
+        qCWarning(lcMkcal) << "empty notebook uid, cannot purge all components.";
+        return false;
+    }
+
+    int rv = 0;
+    int index = 1;
+    const char *query;
+    int qsize = 0;
+    sqlite3_stmt *stmt = nullptr;
+
+    const QByteArray n = notebook.toUtf8();
+
+    if (!d->mDeleteIncComponents) {
+        const char *query = DELETE_COMPONENTS;
+        int qsize = sizeof(DELETE_COMPONENTS);
+        SL3_prepare_v2(d->mDatabase, query, qsize, &d->mDeleteIncComponents, nullptr);
+    }
+
+    query = SELECT_ROWID_FROM_COMPONENTS_BY_NOTEBOOK;
+    qsize = sizeof(SELECT_ROWID_FROM_COMPONENTS_BY_NOTEBOOK);
+
+    SL3_prepare_v2(d->mDatabase, query, qsize, &stmt, NULL);
+    SL3_bind_text(stmt, index, n.constData(), n.length(), SQLITE_STATIC);
+
+    SL3_step(stmt);
+    while (rv == SQLITE_ROW) {
+        int rowid = sqlite3_column_int(stmt, 0);
+
+        int index2 = 1;
```

Comment:
Could use the existing index?

Reply:
I wanted to avoid reusing a variable with the same meaning in a different loop. So I created another one.

Reply:
Fair enough. As more of a temporary variable I wouldn't mind reuse, but however you like.